src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -327,19 +327,23 @@ export const useSwapper = () => {
         const buyAmount = fromBaseUnit(quote.buyAmount, buyAsset.currency.precision)
         const sellAmount = fromBaseUnit(quote.sellAmount, sellAsset.currency.precision)
         const newFiatAmount = bn(buyAmount).times(bnOrZero(buyAsset.fiatRate)).toFixed(2)
+        const estimatedGasFee = fromBaseUnit(quote?.feeData?.fee || 0, 18) // ETH base
```

Comment:
not a fan of 18 magic number for precision here, even with a comment. can you please get the `feeAsset` (ETH) and use the `.precision` from the `feeAsset` here. we already have some sort of selectors/utils to get this given an arbitrary asset.

Reply:
I used the `feeAsset` precision from the `TradeInput` component which I passed down. The issue is that the `feeAsset` can only be set after the `sellAsset` is set, so AFIK there has to be some sort of fallback at some point in the the execution, since we need to handle a race condition. But if there is a way to get the `feeAsset` directly in `useSwapper` let me know